src/osd/scrubber/scrub_machine.cc

```diff
@@ -1149,11 +1020,25 @@ sc::result ReplicaActiveOp::react(const StartReplica&)
   return transit<ReplicaActiveOp>();
 }
 
+
+sc::result ReplicaActiveOp::react(const ReplicaReserveReq& ev)
+{
+  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
+  dout(10) << "ReplicaActiveOp::react(const ReplicaReserveReq&)" << dendl;
+  scrbr->get_clog()->warn() << fmt::format(
+      "osd.{} pg[{}]: reservation requested while processing a chunk",
+      scrbr->get_whoami(), scrbr->get_spgid());
+  // exit-n-renter ReplicaActive (aborting the chunk & freeing the reservation
+  // in the process)
+  post_event(ev);
+  return transit<ReplicaActive>();
```

Comment:
This case is supposed to be impossible, right?  Once again, if it's meant to be impossible it won't be tested and it's not clear to me why this is actually safer.

Reply:
I had the same reasoning as above: I consider freeing the reserver count to be the most important issue for the cluster long-term. And - this reaction is relatively safe.

Regarding testing: yes - that's a valid point. I am working on adding tests to the standalone/scrub collection, with internal debug support to trigger some unexpected scenarios.

Reply:
Do not add code to trigger this (again, impossible) case.  I suppose I'm ok with leaving it since it's supposed to be impossible and releasing state does seem safe-ish, but let's not add even more complexity.

Reply:
You were right: 

there is no way to signal an aborted chunk operation from the replica to the primary (apart from using the 'preempted' flag, which I do not wish to overload with incompatible semantics( * )). Thus, if I use the 'reenter-ReplicaActive' mechanism - the reserver slot is indeed freed, but the primary remains stuck, awaiting for the chunk data. 

A possible behavior would be to do as was already suggested for unexpected backend issues - and take the OSD down. I am hesitant to add this to Squid at this phase. So - yes - let's just release the reserver, allowing other PGs to go on scrubbing.

( * )  for example - if preemption is disabled on the primary side, a 'preempted' response from the replica would be a problem.

Reply:
This comment doesn't need to be addressed in order to merge, but a reason to simply allow mid-chunk reservations is that in it would allow you in the future to change the primary behavior to start the reservation process mid-scrub. There is no correctness dependency here from the replica's point of view, so it seems unnecessarily rigid to impose this requirement.